Update Beatport API search limit and interface parsing

bascurtiz opened the pull request:
Hi Dniel97,

As mentioned before, I made some changes to the Beatport module. 
Can u review? Thanks.

- beatport_api.py: Search limit no longer bound to 10, now uses search_limit from settings.json
- interface.py: Adjusted parsing logic

bascurtiz committed:
Update Beatport API search limit and interface parsing

- beatport_api.py: Search limit no longer bound to 10, now uses search_limit from settings.json

- interface.py: Adjusted parsing logic

bascurtiz committed:
- Cleaned up
- Respecting style conventions
- Ditched future TODO's

bascurtiz commented:
@Orfidev
I've followed up on your suggestions, cleaning up the code, take a look https://github.com/Dniel97/orpheusdl-beatport/pull/28/commits/876599eb8fa660a9b801aa327e190818bc81f8d1

bascurtiz committed:
Fix for issue #25 and #27

bascurtiz committed:
Fix for TypeError: unsupported operand type(s) for //: 'NoneType' and 'int'
Example URL:
https://www.beatport.com/release/the-right-way-mark-maxwell-extended-remix/4639133
Defaults to 0 if a track's length_ms is None

bascurtiz committed:
Adds playlist download support (this breaks with original OrpheusDL, so use my
fork instead)

bascurtiz committed:
Backward compatibility fix for og OrpheusDL (No more relying on utils/models.py)

bascurtiz committed:
Updated README.md

bascurtiz committed:
Resolving AttributeError issues by making all chained .get() calls safe to handle None values

bascurtiz committed:
Add pre-download validation to catch corrupted files and prevent tagging errors

bascurtiz committed:
Updated README.md

bascurtiz committed:
Fix: Handle region-locked tracks gracefully instead of crashing

bascurtiz committed:
Concurrency downloads added

Dniel97 commented on interface.py:
```diff
@@ -285,7 +285,7 @@ def get_playlist_info(self, playlist_id: str, is_chart: bool = False) -> Playlis
                 logging.warning(f"Stopped pagination for {'chart' if is_chart else 'playlist'} {playlist_id} at page {current_page}. Expected {total_items}, got {len(all_tracks_raw)}.")
                 break
-        if total_items > 0: self.print("") # Clear the progress line by printing a newline
+
```
Why did you remove that?

bascurtiz replied:
I thought they would interfere with how i have the output changed in my fork of orpheusdl, but it seems it doesn't have impact. Will delete comit.

Dniel97 commented on interface.py:
```diff
@@ -374,7 +374,6 @@ def get_artist_info(self, artist_id: str, get_credited_albums: bool, is_chart: b
         artist_tracks = artist_tracks_data.get("results")
         total_tracks = artist_tracks_data.get("count")
         for page in range(2, total_tracks // 100 + 2):
-            print(f"Fetching {page * 100}/{total_tracks}", end="\r")
```
Why did yu remove that as well?

bascurtiz replied:
I thought they would interfere with how i have the output changed in my fork of orpheusdl, but it seems it doesn't have impact. Will delete comit.

Dniel97 commented on interface.py:
```diff
@@ -575,10 +584,11 @@ def get_track_download(self, track_id: str, quality_tier: QualityEnum) -> TrackD
-                raise self.exception(f"Track '{track_id}' does not contain valid audio content")
+            # Only check for obviously non-audio content types (like HTML error pages)
+            # Be more permissive with content type validation since audio can be served with various MIME types
+            if content_type and any(bad_type in content_type.lower() 
```
No, I would rather check for the actual types that I want than checking types I don't expect.

bascurtiz committed:
Delete interface.py

bascurtiz committed:
Delete beatport_api.py

bascurtiz committed:
Add files via upload

bascurtiz committed:
Fucking up reverting

bascurtiz committed:
Merge branch 'master' of https://github.com/bascurtiz/orpheusdl-beatport

bascurtiz committed:
follow up to https://github.com/Dniel97/orpheusdl-beatport/pull/28

bascurtiz committed:
Fix for playlist support

bascurtiz committed:
Updated interface.py to allow 'video' and 'mp4' content types for audio streams

bascurtiz committed:
- Added image and preview URL extraction to search results (with artist image fallback)

bascurtiz committed:
Improve error messages and add upfront credential validation to prevent unclear errors

bascurtiz committed:
feat: add Beatport label support and API improvements

- Add get_label_releases/get_label_tracks with pagination in beatport_api
- Parse /label/ URLs and implement get_label_info in interface
- Support labels in search results and URL parsing
- Fix artist track pagination, add str/int key handling for cached track/album data
- Add track preview_url and id to TrackInfo
- Show track count for releases/playlists, filter empty playlists

bascurtiz committed:
fix: Beatport auth session recovery and error handling

- valid_account(): on subscription/link errors, clear tokens and re-login
- refresh_login(): on any refresh failure, clear session and re-login (not just invalid_grant)
- Album fetch: handle 404 "not found" explicitly; ensure all errors are logged

bascurtiz committed:
- Re-login on subscription/link errors and any refresh failure
- Handle 404 for missing releases; always surface album errors

bascurtiz committed:
Less strict check to verify if it's a pro account (bp_link_pro vs bp_link_pro_annual for ex.)

bascurtiz committed:
feat: map ATMOS quality to medium

bascurtiz committed:
feat(beatport): add anonymous mode support and improve search fallback
- Implement automatic anonymous token fetching when no credentials are provided.
- Added logic to run in anonymous mode if username/password are missing
- Refactored search to fall back to general search when using anonymous tokens
- Added requirement for logged-in account for track downloads
- Updated User-Agent to a modern version
- Cleaned up debug logging and imports in beatport_api.py
- Improved error handling for 401 and 403 status codes

bascurtiz committed:
Letting user know anonymous mode is limited in search results

bascurtiz committed:
feat: improve pagination, error handling, and performance
- Add pagination support to artist and label catalog endpoints
- Implement multi-threaded batch fetching for release metadata and durations
- Enhance error handling for region locks and API inconsistencies
- Add BPM and mix name to track metadata
- Fix potential crashes with missing cover URLs
- Improve anonymous search by refreshing tokens automatically

bascurtiz committed:
Refined token extraction logic to be more targeted and added better error logging

bascurtiz committed:
feat: optimize artist discography fetching and improve robustness
- Reuse already fetched durations for releases in artist info
- Increase threading concurrency (max_workers=10)
- Add safety checks for API responses and metadata parsing
- Cap batch fetching at 250 releases to avoid rate limits
